core/src/test/java/kafka/metrics/ClientMetricsInstanceTest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.metrics;
+
+import org.apache.kafka.common.Uuid;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.net.UnknownHostException;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class ClientMetricsInstanceTest {
+
+    private Uuid uuid;
+    private ClientMetricsInstanceMetadata instanceMetadata;
+    private ClientMetricsInstance clientInstance;
+
+    @BeforeEach
+    public void setUp() throws UnknownHostException {
+        uuid = Uuid.randomUuid();
+        instanceMetadata = new ClientMetricsInstanceMetadata(uuid,
+            ClientMetricsTestUtils.requestContext());
+        clientInstance = new ClientMetricsInstance(Uuid.randomUuid(), instanceMetadata, 0, 0,
+            null, ClientMetricsConfigs.DEFAULT_INTERVAL_MS);
+    }
+
+    @Test
+    public void testMaybeUpdateRequestEpochValid() {
+        // First request should be accepted.
+        assertTrue(clientInstance.maybeUpdateGetRequestEpoch(System.currentTimeMillis()));
```

Comment:
Could we use MockTime instead of System time?

Reply:
As the class/methods I am testing do not have `Time` reference rather just tries to updates the epoch supplied hence I didn't see any value in using MockTime. However I did change the code and tests for ClientMetricsManager where these methods are invoked from, I started using Time in ClientMetricsManager and corresponding MockTime in ClientMetricsManagerTest which eliminated the use of `Thread.sleep`.

Please let me know if I am missing anything here.